### ISSUE TITLE
Cannot modify HTTPScaledObjects after moving from 0.7.0 to 0.8.0

### ISSUE THREAD

**Issue by jocelynthode** [kedacore/http-add-on#1020]
### Report

After upgrading to 0.8.0. The CRD seems to have changed and the enum for `status.conditions[].type` now only supports `Ready` as a value.

Before this update to the CRD we supported the following values:
```
- Created
- Error
- Pending
- Unknown
- Terminating
- Terminated
- Ready
```

I was able to workaround it by editing the CRD and readding these statuses and then edit them so that the status conditions could be cleaned up.

### Expected Behavior

I expect to be able to edit old HTTPScaledObjects

### Actual Behavior

Old `HTTPScaledObjects` fail to be edited with the following messages:
```
* status.conditions[414].type: Unsupported value: "Pending": supported values: "Ready"
* status.conditions[415].type: Unsupported value: "Created": supported values: "Ready"
* status.conditions[417].type: Unsupported value: "Pending": supported values: "Ready"
* status.conditions[418].type: Unsupported value: "Created": supported values: "Ready"
* status.conditions[420].type: Unsupported value: "Pending": supported values: "Ready"
* status.conditions[421].type: Unsupported value: "Created": supported values: "Ready"
* status.conditions[423].type: Unsupported value: "Pending": supported values: "Ready"
* status.conditions[424].type: Unsupported value: "Error": supported values: "Ready"
```

### Steps to Reproduce the Problem

1. Have an old HTTPScaledObject with some status conditions other than Ready
2. Update to 0.8.0
3. Try to edit the HTTPScaledObjects

### HTTP Add-on Version

0.8.0

### Kubernetes Version

1.27

### Platform

Other

**Comment by massimeddu-sj** [comment]
Same issue here on Kubernetes 1.29

**Comment by JorTurFer** [comment]
f**k I didn't notice that changing the status fields I can break the CRD :/ 
The HTTP Add-on shouldn't use them anymore either but probably it doesn't remove them if they are already there. Does it fail after the rollout of the new version is completed?

**Comment by JorTurFer** [comment]
I'm asking because we have to include a check for removing those states from the HTTPScaledObject if they are available

**Comment by jocelynthode** [comment]
> f**k I didn't notice that changing the status fields I can break the CRD :/ The HTTP Add-on shouldn't use them anymore either but probably it doesn't remove them if they are already there. Does it fail after the rollout of the new version is completed?

Yes The upgrade to 0.8.0 was successful and everything seemed to be working. Then when I wanted to edit `HTTPScaledObjects` created before the upgrade I was met with the error. The problem is indeed that it does not remove old statuses and thus after the upgrades prevents objects with old statuses to be updated.

**Comment by JorTurFer** [comment]
> Yes The upgrade to 0.8.0 was successful and everything seemed to be working. Then when I wanted to edit `HTTPScaledObjects` created before the upgrade I was met with the error. The problem is indeed that it does not remove old statuses and thus after the upgrades prevents objects with old statuses to be updated.

Thanks for the feedback, I'm going to try to fix this during the week and cut a fix release. I thought that old values were dropped automatically but apparently not (every day you learn things xD). The operator will drop them during the upgrade for future releases. Sorry for the inconvenience

**Comment by jocelynthode** [comment]
No worries, thanks for the work :slightly_smiling_face:

**Comment by oto-macenauer-absa** [comment]
just encountered this, bumping so it doesn't get stale:)

**Comment by jocelynthode** [comment]
@JorTurFer : Any update on this issue? It's been a few months with no releases.

**Comment by jocelynthode** [comment]
Do not close

**Comment by jegork** [comment]
Still an issue
